src/src/pages/capabilities/KafkaCluster/NewTopicDialog.js

```diff
@@ -58,7 +58,11 @@ export default function NewTopicDialog({
   const changeRetention = (e) => {
     e.preventDefault();
     const newValue = e?.target?.value || emptyValues.retention;
-    setFormData((prev) => ({ ...prev, ...{ retention: parseInt(newValue) } }));
+    let finalValue = newValue;
+    if (!isNaN(finalValue)) {
+      finalValue = parseInt(newValue);
+    }
+    setFormData((prev) => ({ ...prev, ...{ retention: finalValue } }));
```

Comment:
What happens in the case of a NaN value then? The flow is a bit hard to read since you are also reusing the same variable. Do we have a fall back in case of bad data from e?

Reply:
In the case of a NaN value (so not a number) unless someone changes our predefined values, it can only be a string value with "forever".

Reply:
Ah now I get it. Can you do comparison with "forever" instead?

Reply:
Will do. It does mean with that change it could potentially at some point try to run parseInt for a string that wouldn't be considered a number, whereas the current check would cover that.